### DISCUSSION TITLE
How can I reload page without flickering?

### DISCUSSION THREAD

**Discussion by leesb971204** [TanStack/router#2989] [Q&A]
I am trying to convert the following react-router-dom v5 code to tanstack-router v1:
```typescript
const location = useLocation();
const history = useHistory();
history.replace(location.pathname);
``` 

This logic is used in my code to refresh the page without causing any flickering. Specifically, it is intended to update cookies while ensuring a seamless refresh experience without a visible page reload.

Here is my attempt at implementing the same functionality using tanstack-router:
```typescript
const router = useRouter();
const navigate = useNavigate();

// 1
router.history.push(`${location.pathname}`, new Date(), true);

// 2
router.history.replace(location.pathname, new Date(), true);

// 3
navigate({ to: location.pathname, replace: true });
``` 

None of these attempts (1, 2, or 3) seem to trigger the cookie refresh. Even when I fix the argument to location.pathname, the result is the same.

Am I missing something here? Any advice or insights would be greatly appreciated.

**Comment by schiller-manuel** [comment]
can you please create a minimal complete example of what you have tried by forking one of the existing stackblitz examples?

**Reply by leesb971204** [reply]
I created a simple project in Stackblitz, but it seems like cookies are not being set in the developer tools. I feel like this doesn’t fully explain my situation. Are there any alternatives?

I just want to know how to write the equivalent of `history.replace(location.pathname);` from react-router-dom in tanstack-router.

**Reply by schiller-manuel** [reply]
you can also provide a github repo

**Reply by leesb971204** [reply]
I create repo
https://github.com/leesb971204/tanstack-router-test

I want the handleClick function in this code to refresh the page without flickering and update the cookie value in the developer tools using `router.history.replace`. This worked in react-router-dom, but it doesn’t work in this code

For reference, not only `router.history.replace(location.pathname);` but also the three methods I mentioned in the text do not work as intended.

**Reply by leesb971204** [reply]
any update?

**Reply by schiller-manuel** [reply]
I tried your example, but when I click on `set 'test' cookie to 1` it does not flicker. however, I guess you need to use:

```tsx
const navigate = useNavigate()
navigate({to: '/'})

```

**Reply by leesb971204** [reply]
It still doesn’t work the way I want it to..
When the function is executed, the cookies in the developer tools should be refreshed.
